### PR TITLE
Add Release entity

### LINKED ISSUE #134: Add Release as an entity that would be inspected and stored
As SrcOpsMetrics user,

I would like to have info about releases of a project so that I can analyze the behaviour after all release.

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
Fixes #134 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description

Analysis of releases for github repositories. Tags are used instead of `release` (as far as I know, we do not even use github's releases anywhere in thoth-station).

### CODE REVIEW

**Comment by pacospace** [comment]
> ## Related Issues and Dependencies
> Fixes #134
> 
> ## This introduces a breaking change
> * [ ]  Yes
> * [x]  No
> 
> ## Description
> Analysis of releases for github repositories. Tags are used instead of `release` (as far as I know, we do not even use github's releases anywhere in thoth-station).

Yes tags should be the good way to go for the releases as we use aicoe-ci and therefore a tag will be created by Sesheta once the release is done and that triggers the creation of the image on quay. @harshad16 might have more thoughts here :)

File: srcopsmetrics/entities/release.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from github.GitTag import GitTag
+from semver import VersionInfo
+from voluptuous.schema_builder import Schema
+
+from srcopsmetrics.entities import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Release(Entity):
```
Comment (pacospace) [inline srcopsmetrics/entities/release.py:33]: This class need to be added to EntityTypeEnum to be used with `-e` in CLI

**Comment by pacospace** [comment]
I made the following test after adding Release to the enumtype class but received an error:

```
PYTHONPATH=. pipenv run python3 srcopsmetrics/cli.py --repository thoth-station/mi -c -l -e Release    
INFO:srcopsmetrics.github_knowledge:Overall repositories found: 1
INFO:srcopsmetrics.bot_knowledge:######################## Analysing thoth-station/mi ########################

INFO:srcopsmetrics.bot_knowledge:Release inspection
INFO:srcopsmetrics.entities.tools.storage:Loading knowledge from Ceph
INFO:srcopsmetrics.entities.interface:No previous knowledge of type Release found
INFO:srcopsmetrics.iterator:-------------Release Analysis-------------
INFO:srcopsmetrics.iterator:Cached knowledge could not be saved
Traceback (most recent call last):
  File "srcopsmetrics/cli.py", line 129, in <module>
    cli()
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "srcopsmetrics/cli.py", line 111, in cli
    analyse_projects(projects=tupled_repos, is_local=is_local, entities=entities)
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/bot_knowledge.py", line 87, in analyse_projects
    github_repo=github_repo, project_path=project_path, entity_cls=entity, is_local=is_local
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/github_knowledge.py", line 187, in analyse_entity
    analysis.run()
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/iterator.py", line 69, in run
    for idx, entity in enumerate(self.entity.analyse(), 1):
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/entities/release.py", line 40, in analyse
    return [tag for tag in self.get_raw_github_data() if tag.name not in self.previous_knowledge]
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/entities/release.py", line 73, in get_raw_github_data
    releases.extend([t for t in self.get_tags()])
AttributeError: 'Release' object has no attribute 'get_tags'
```

File: srcopsmetrics/entities/release.py (hunk 1 above)
Comment (xtuchyna) [reply srcopsmetrics/entities/release.py:33]: no need for that, with the new dynamic import the class can be used with -e option
(see this here https://github.com/thoth-station/mi/pull/186#discussion_r501320326 )

**Comment by xtuchyna** [comment]
@pacospace hmm, dunno why pre-commit is not being done, but it should be aight

**Comment by harshad16** [comment]
@harshad16

File: srcopsmetrics/entities/__init__.py [hunk 2]
```diff
@@ -16,6 +16,6 @@
 # along with SrcOpsMetrics.  If not, see <http://www.gnu.org/licenses/>.
 
 """Entities imports."""
+from .interface import Entity
```
Comment (pacospace) [inline srcopsmetrics/entities/__init__.py:19]: import needed?

Comment (pacospace) [reply srcopsmetrics/entities/__init__.py:19]: please solve pre-commit :)

Comment (xtuchyna) [reply srcopsmetrics/entities/__init__.py:19]: absolutelly forgot about this one
import is needed as everything else besides `entities` module uses only Entity import alias `from entities import Entity`
i've just added it to __all__ and seems like pre-commit is fine w/ it

**Review by pacospace — APPROVED** [review]
Thanks!!

**Pull request merged**